File: admin/src/app/Dtos/order.ts
import { Facility } from './facility';
import { Hotel } from './hotel';
import { Room } from './room';

export interface Order {
    room: Room;
    hotel: Hotel;
    checkInDate: string;
    checkOutDate: string;
    facilities: Facility[];
    price: number;
}

export interface OrderToAdd {
    hotelId: number;
    roomId: number;
    facilities: Facility[];
    checkInDate: string;
    checkOutDate: string;
}

File: api/Models/Order.cs
using System.Collections.Generic;

namespace iTechArt.Hotels.Api.Models
{
    public class Order
    {
        public Hotel Hotel { get; set; }
        public Room Room { get; set; }
        public decimal Price { get; set; }
        public List<Facility> Facilities { get; set; }
        public string CheckInDate { get; set; }
        public string CheckOutDate { get; set; }
    }
}
